jaeger-thrift/src/main/java/io/jaegertracing/thrift/internal/senders/HttpSender.java

```diff
@@ -119,13 +134,68 @@ public Builder withAuth(String authToken) {
       return this;
     }
 
+    public Builder withCertificatePinning(String[] sha256certs /* comma separated */) {
+      pins.addAll(Arrays.asList(sha256certs));
+      return this;
+    }
+
+    /**
+     * Enable accepting self-signed certificates. This will only take effect if pinning is provided.
+     */
+    public Builder acceptSelfSigned() {
+      this.selfSigned = true;
+      return this;
+    }
+
     public HttpSender build() {
       if (authInterceptor != null) {
         clientBuilder.addInterceptor(authInterceptor);
       }
+      try {
+        // Just obtain hostname for SSL feature. Failure is OK if plain http is used.
+        final URI uri = new URI(endpoint);
+        String hostname = hostname = uri.getHost();
+
+        if (uri.getScheme() != null && uri.getScheme().equals("https")) {
+          if (!selfSigned && !pins.isEmpty()) {
+            // Pinning Certificate issued by public CA
+            for (String cert: pins) {
+              certificatePinnerBuilder.add(hostname, String.format("%s", cert));
+            }
+            clientBuilder.certificatePinner(certificatePinnerBuilder.build());
+          } else if (selfSigned && !pins.isEmpty()) {
+            /* Issued by private CA, OkHttp's pinner is unable to verify the pins.
+             * Instead, check the sha256 hash value by custom verifier. */
+            acceptSelfSigned(clientBuilder, hostname, pins);
+          }
+        }
+      } catch (java.net.URISyntaxException e) {
+        // Early but similar validation & exception as what happens when HttpSender is called.
+        throw new IllegalArgumentException("Could not parse endpoint.", e);
+      }
       return new HttpSender(this);
     }
 
+    private void acceptSelfSigned(OkHttpClient.Builder clientBuilder, final String hostname, final List<String> pinlist) {
+      try {
+        final TrustManager[] selfSignedServerTrustManager = new TrustManager[] { new SelfSignedTrustManager(hostname, pinlist) };
+        final SSLContext sslContext = SSLContext.getInstance("TLS");
+        sslContext.init(null, selfSignedServerTrustManager, null);
+        clientBuilder.sslSocketFactory(sslContext.getSocketFactory(), (X509TrustManager) selfSignedServerTrustManager[0]);
+
+      } catch (java.security.NoSuchAlgorithmException e) {
+          // TLS is hardcoded abave. No occasion to come here.
+          throw new RuntimeException(e);
+      } catch (java.security.KeyManagementException e) {
```

Comment:
Any reasons not to import the class, instead of referring it to its fully qualified name?